lib/datasource/nuget/index.ts

```diff
@@ -6,17 +6,24 @@ import { GetReleasesConfig, ReleaseResult } from '../common';
 
 export { id } from './common';
 
-function detectFeedVersion(url: string): 2 | 3 | null {
+function parseRegistryUrl(
+  registryUrl: string
+): { feedUrl: string; protocolVersion: number } {
```

Comment:
A question, but not a request: Is it worth specifying the version as a type here. `protocolVersion: 2 | 3` as before, and `protocolVersionRegExp.exec(parsedUrl.hash) as 2 | 3`.

Reply:
I also thought about that, but decided against it for readability/simplicity.